index.js

```diff
@@ -250,6 +239,17 @@ function SonicBoom (opts) {
     }
   })
 }
+function releaseWritingBuf (writingBuf, len, n) {
+  // if writingBuf.length is equal to n, that means writingBuf contains no multi-byte character
+  if (typeof writingBuf === 'string' && writingBuf.length !== n) {
+    // Since the fs.write callback parameter `n` means how many bytes the passed of string
+    // We calculate the original string length for avoiding the multi-byte character issue
+    n = Buffer.from(writingBuf).subarray(0, n).toString().length
+  }
+  len = Math.max(len - n, 0)
+  writingBuf = writingBuf.slice(n)
+  return { writingBuf, len }
```

Comment:
Could most of this be solved by comparing `theBuffer.byteLength` instead of `theBuffer.length`?

Reply:
Thank you. Using `.byteLength` is more sensible to compare byteLength.